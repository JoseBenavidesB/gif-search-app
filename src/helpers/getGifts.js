export const getGifts = async (category)=> {


    const key = 'NXVy4RhhTbYjqhhu0L5wyXK0d1xYD5W2'
    const url = `https://api.giphy.com/v1/gifs/search?api_key=${key}&q=${category}&limit=5`;

    const resp = await fetch(url);
    const {data} = await resp.json();

    const gifs = data.map((img)=> {
        return {
            id: img.id,
            title: img.title,
            url: img.images?.downsized_medium.url //? existe?

        }
    });
    //console.log(gifs)
    
    return gifs;


};