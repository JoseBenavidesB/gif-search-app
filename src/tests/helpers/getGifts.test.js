import { getGifts } from '../../helpers/getGifts';

describe('Test getGifts', ()=>{

    test('should return array with 5', async()=> {

        const gifs = await getGifts('one punch')
        
        expect( gifs.length ).toBe( 5 );
    });

    test('should return  array with 0', async()=> {

        const gifs = await getGifts('')
        
        expect( gifs.length ).toBe( 0 );
    });
});
