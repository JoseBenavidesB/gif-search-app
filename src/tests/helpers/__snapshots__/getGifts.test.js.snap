// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <getGifts/> should return <getGifts/> correctly 1`] = `<getGifts />`;
